src/sql/workbench/parts/dataExplorer/common/nodeContext.ts

```diff
@@ -76,4 +92,13 @@ export class NodeContextKey extends Disposable implements IContextKey<INodeConte
 	get(): INodeContextValue | undefined {
 		return this._nodeContextKey.get();
 	}
+
+	private isConnectableNode(nodeContextValue: INodeContextValue): boolean {
```

Comment:
Something is connectable if we get a connectionprofile as a payload for an item from the tree providers extension. Or if we know from using the objectExplorerShim class. I don't think we need is database and is server from the extension as well. Is we need to specify between databases and servers we can use the existing "contextValue" that vscode's tree already has and heavily recommend extensions use a generic value for this like "server" or "database".

Reply:
the contextValue right now are just strings with Regex logic to look for "database" or "server". Wouldn't it make sense to have them as objects instead of strings and looking for them?

Reply:
Similar to what karl said, this api is generic, extensions can put anything in there, and in theory we should support extensibility in any case, so strings are the only real option. Regex is already how extensions interact with trees throughout trees in vscode currently, so it's not against the current models.

Reply:
Maybe I don't understand correctly, but how's the current implementation different? What stops an extension to have a node that has "server" or "database" in its contextValue to show incorrect context menu actions?
But if VSCode is using this method, I guess I can revert it to be based on strings.